app_3/views.py

```diff
@@ -0,0 +1,31 @@
+from django.shortcuts import render
+from datetime import datetime, timedelta
+from .models import User, Order
+
+
+def index(request):
+    return render(request, 'app_3/index.html')
+
+
+def cart(request, user_id: int):
+    products = []
+    user = User.objects.filter(pk=user_id).first()
+    orders = Order.objects.filter(customer=user).all()
+    for order in orders:
+        products.append(order.products.all())
+    products.reverse()
+    return render(request, 'app_3/all_orders.html', {'user': user, 'orders': orders, 'products': products})
```

Comment:
Например можно сделать так
```suggestion
    products = {}
    user = User.objects.filter(pk=user_id).first()
    orders = Order.objects.filter(customer=user).all()
    for order in orders:
        products[order] = order.products.all()
    print(products)
    return render(request, 'app_3/all_orders.html', {'user': user, 'products': products})
```